code/modules/mob/holder.dm

```diff
@@ -43,58 +60,220 @@
 	if(!holder_type || buckled || pinned.len)
 		return
 
+	if ((grabber.hand == 0 && grabber.r_hand) || (grabber.hand == 1 && grabber.l_hand))//Checking if the hand is full
+		grabber << "Your hand is full!"
+		return
+
+
 	var/obj/item/weapon/holder/H = new holder_type(loc)
 	src.loc = H
 	H.name = loc.name
 	H.attack_hand(grabber)
+	H.contained = src
 
 	grabber << "You scoop up [src]."
 	src << "[grabber] scoops you up."
 	grabber.status_flags |= PASSEMOTES
 	return
 
 //Mob specific holders.
+//w_class mainly determines whether they can fit in trashbags. <=2 can, >=3 cannot
 /obj/item/weapon/holder/diona
 	name = "diona nymph"
-	desc = "It's a tiny plant critter."
+	desc = "It's a little plant critter."
 	icon_state = "nymph"
 	origin_tech = "magnets=3;biotech=5"
 	slot_flags = SLOT_HEAD | SLOT_OCLOTHING
+	w_class = 2
+
+/obj/item/weapon/holder/diona/dead
+	name = "diona nymph"
+	desc = "It used to be a little plant critter"
+	icon_state = "nymph_dead"
+	origin_tech = "magnets=3;biotech=5"
+	slot_flags = null
+	w_class = 2
+
 
 /obj/item/weapon/holder/drone
 	name = "maintenance drone"
 	desc = "It's a small maintenance robot."
 	icon_state = "drone"
 	origin_tech = "magnets=3;engineering=5"
+	slot_flags = SLOT_HEAD
+	w_class = 2
 
 /obj/item/weapon/holder/cat
 	name = "cat"
 	desc = "It's a cat. Meow."
-	icon_state = "cat"
+	icon_state = "cat_tabby"
 	origin_tech = null
+	slot_flags = SLOT_HEAD
+	w_class = 3
+
+/obj/item/weapon/holder/cat/dead
+	name = "cat"
+	desc = "It's a dead cat."
+	icon_state = "cat_tabby_dead"
+	origin_tech = null
+	slot_flags = null
```

Comment:
Whenever you see bitfields, which this is, please use `0` instead of `null`. Any bitfield operations trying to interface with null may and should runtime, where as 0 will get the desired results.